fondat/aws/bedrock/resources/streams.py

```diff
@@ -53,6 +53,66 @@ async def __aexit__(
     ):
         await self.close()
 
+    # ------------- public helper -------------
+    async def close(self):
+        """Close the underlying aiohttp session if not already closed."""
+        if not self._closed:
+            await self._client_cm.__aexit__(None, None, None)
+            self._closed = True
+
+
+class AgentStream(AsyncIterator[dict]):
```

Comment:
Why duplicate code like this ? Can't we use a `BaseStream`  and subclass from there ?
```
class BaseStream(AsyncIterator[dict]):

.... common code ....

class FlowStream(BaseStream):
    """
    Async iterator for flow streams from invoke_flow.
    """

    def __init__(self, response: dict, client_cm):
        super().__init__(response, client_cm, "responseStream")


class AgentStream(BaseStream):
    """
    Async iterator for agent completion streams from invoke_agent.
    """

    def __init__(self, response: dict, client_cm):
        super().__init__(response, client_cm, "completion")
```